### PR TITLE
Create github action for python testing and pypi publishing

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- Add github action for python packages pytest
- Add github action for publishing pypi artifacts

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Regenerate TrainedModel SDK after downgrading to v1alpha1
2. Add github action for python test CI and remove from workflow

**Release note**:

```release-note
KFServing python SDK should be automatically published when tags are created
```

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
```
Comment (jinchihe) [inline .github/workflows/python-publish.yml:25]: I wonder where the env set? Thanks.

**Review by jinchihe — COMMENTED** [review]
That's looks great @yuzisun please resolve conflicts.

**Comment by jinchihe** [comment]
@yuzisun for each publishing, the version in setup.py should be updated with the tag? right?

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (yuzisun) [reply .github/workflows/python-publish.yml:25]: github settings can configure the secrets and they are encrypted, the value can be referenced like above with the configured variable names.

**Comment by yuzisun** [comment]
> @yuzisun for each publishing, the version in setup.py should be updated with the tag? right?

the version in setup.py should match the release tag, you updated the version in setup.py last time to [0.5.0](https://github.com/kubeflow/kfserving/blob/master/python/kfserving/setup.py#L28),

**Comment by jinchihe** [comment]
OK， Thanks

/lgtm

**Pull request merged**